torchao/dtypes/nf4tensor.py

```diff
@@ -37,11 +40,234 @@ def decorator(func):
     return decorator
 
 
-@implements([torch.ops.aten.detach.default, torch.ops.aten.detach])
+@implements([torch.ops.aten.detach])
 def noop_detach(func, *args, **kwargs):
     return args[0][0]
 
 
+@implements(
+    [
+        aten.detach.default,
+    ]
+)
+def nf4_detach(aten_op, args, kwargs=None):
+    # nn.Parameter need detach
+    quantized_scalers = aten_op(args[0].quantized_scalers, *args[1:], **kwargs)
+    quantization_factor = aten_op(args[0].quantization_factor, *args[1:], **kwargs)
+    quantized_data = aten_op(args[0].quantized_data, *args[1:], **kwargs)
+    scaler_mean = aten_op(args[0].scaler_mean, *args[1:], **kwargs)
+    nf4 = aten_op(args[0].nf4, *args[1:], **kwargs)
+    tensor_meta = SubclassTensorArgs(
+        args[0].size(),
```

Comment:
nit, maybe make a param: NF4tensor = args[0] ...

Reply:
it feels like something neat. but i did not fully understand your suggestion yet

Reply:
+1 This also confused me. I think what driss means is just give a human readable name to args[0] so its easier to read the code

Reply:
declared `nf4tensor = args[0]` at the begining so it becomes easier to understand